src/Beberlei/Bundle/MetricsBundle/DependencyInjection/Configuration.php

```diff
@@ -18,21 +18,40 @@ public function getConfigTreeBuilder()
 
         $rootNode
             ->children()
-                ->scalarNode('default')->defaultValue('default')->end()
+                ->scalarNode('default')
+                    ->defaultNull()
+                ->end()
                 ->arrayNode('collectors')
                     ->isRequired()
                     ->useAttributeAsKey('name')
                     ->prototype('array')
                         ->children()
-                            ->scalarNode('type')->isRequired()->end()
-                            ->scalarNode('hostname')->end()
-                            ->scalarNode('server')->end()
-                            ->scalarNode('host')->end()
-                            ->scalarNode('port')->end()
-                            ->scalarNode('file')->end()
-                            ->scalarNode('username')->end()
-                            ->scalarNode('password')->end()
-                            ->scalarNode('connection')->end()
+                            ->scalarNode('type')->end()
+                            ->scalarNode('connection')->defaultNull()->end()
+                            ->scalarNode('file')->defaultNull()->end()
+                            ->scalarNode('host')->defaultNull()->end()
+                            ->scalarNode('password')->defaultNull()->end()
+                            ->scalarNode('port')->defaultNull()->end()
+                            ->scalarNode('prefix')->defaultNull()->end()
+                            ->scalarNode('protocol')->defaultNull()->end()
+                            ->scalarNode('source')->defaultNull()->end()
+                            ->scalarNode('username')->defaultNull()->end()
+                        ->end()
+                        ->validate()
+                            ->ifTrue(function ($v) { return 'librato' === $v['type'] && empty($v['source']); })
+                            ->thenInvalid('The source has to be specified to use a Librato')
+                        ->end()
+                        ->validate()
+                            ->ifTrue(function ($v) { return 'librato' === $v['type'] && empty($v['username']); })
+                            ->thenInvalid('The username has to be specified to use a Librato')
+                        ->end()
+                        ->validate()
+                            ->ifTrue(function ($v) { return 'librato' === $v['type'] && empty($v['password']); })
+                            ->thenInvalid('The password has to be specified to use a Librato')
+                        ->end()
+                        ->validate()
+                            ->ifTrue(function ($v) { return 'doctrine_dbal' === $v['type'] && empty($v['connection']); })
```

Comment:
see my previous comment. the default DBAL connection is aliased as ``database_connection`` in the container. This is the service you should reference when getting ``null``